docs/src/pages/components/data-grid/rows/ServerInfiniteLoadingGrid.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { GridFetchRowsParams, GridRowData, XGrid } from '@material-ui/x-grid';
+import {
+  useDemoData,
+  getRealData,
+  getCommodityColumns,
+} from '@material-ui/x-grid-data-generator';
+
+async function sleep(duration) {
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, duration);
+  });
+}
+
+const loadServerRows = async (newRowLength: number): Promise<GridRowData[]> => {
+  const newData = await getRealData(newRowLength, getCommodityColumns());
+  // Simulate network throttle
+  await sleep(Math.random() * 100 + 100);
+
+  return newData.rows;
+};
+
+export default function InfiniteLoadingGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 10,
+    maxColumns: 6,
+  });
+
+  const handleFetchRows = async (params: GridFetchRowsParams) => {
+    const newRowsBatch: GridRowData[] = await loadServerRows(
+      params.viewportPageSize,
+    );
+
+    params.api.current.insertRows({
+      startIndex: params.startIndex,
+      pageSize: params.viewportPageSize,
+      newRows: newRowsBatch,
+    });
```

Comment:
This looks wrong. The data grid is the best one to know if the resolved value is still relevant or not, not the developer at this level. I think that we should simply yield:

```suggestion
    return {
      startIndex: params.startIndex,
      pageSize: params.viewportPageSize,
      newRows: newRowsBatch,
    };
```

At the very least, if we really want to keep this API, we need to update this demo to force the requests to resolve in the right order (ignore outdated requests). Which we don't do here.

Reply:
The initial version was like the one you are suggesting but after discussing it with @dtassone I reworked it to match the rest of the grid's APIs. In that case the responsibility of loading the rows falls on to the developer.

Reply:
In this case, I think that it's important we update the demo to showcase a sound implementation of data fetching. It will be a good forcing function to see the complexity we are pushing on developers, it will help us evaluate if we are happy with this or note.

Reply:
I guess this is what you mean https://github.com/mui-org/material-ui-x/pull/1715#issuecomment-861922839. The current demo does show how to use the feature in action, it just doesn't do the filtering and sorting part.